### PR TITLE
Initial implementation of Amplitude service

### PR DESCRIPTION
**Pull request by darinkrauss**
- Implement Service and ServiceUI protocols
- Implement Analytics protocol
- Connect with Amplitude third-party service

https://tidepool.atlassian.net/browse/LOOP-22

### CODE REVIEW

File: AmplitudeServiceKit/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/* The title of the Amplitude service */
+"Amplitude" = "Amplitude";
```
Comment (nhamming) [inline AmplitudeServiceKit/en.lproj/Localizable.strings]: Maybe I'm mistaken, but won't the default language localization be provided when a `.strings` file is missing. If yes, I don't see the point of making `.strings` files that do not provide any localization.

File: AmplitudeServiceKitTests/AmplitudeServiceKitTests.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//  AmplitudeServiceKitTests
+//
+//  Created by Darin Krauss on 6/18/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import XCTest
+@testable import AmplitudeServiceKit
+
+class AmplitudeServiceKitTests: XCTestCase {
```
Comment (nhamming) [inline AmplitudeServiceKitTests/AmplitudeServiceKitTests.swift]: Should this be committed?

File: AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.register(AuthenticationTableViewCell.nib(), forCellReuseIdentifier: AuthenticationTableViewCell.className)
```
Comment (nhamming) [inline AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift:39]: Just want to flag the `.nib()` since I recently saw a change to `.self`.

**Review by nhamming — DISMISSED** [review]
Otherwise LGTM

File: AmplitudeServiceKit/en.lproj/Localizable.strings (hunk 1 above)
Comment (darinkrauss) [reply AmplitudeServiceKit/en.lproj/Localizable.strings]: Yes, good call. Will remove in the three service repos.

File: AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift (hunk 3 above)
Comment (darinkrauss) [reply AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift:39]: The use of `.self` or `.nib()` depends upon whether the view is implemented strictly in code (where `.self` would be used) or whether it uses a `.xib` file (where `.nib()` would be used). 

In the case of the recent change from `.nib()` to `.self` for `SwitchTableViewCell`, the `.xib` file was recently deleted and the view implemented only in code. See 9b23b8330f3507aa39ccd3f716b556b033f3c8ba for details.

In the above case, the `AuthenticationTableViewCell` uses a `.xib` file so the use of `.nib()` is correct.

File: AmplitudeServiceKitTests/AmplitudeServiceKitTests.swift (hunk 2 above)
Comment (darinkrauss) [reply AmplitudeServiceKitTests/AmplitudeServiceKitTests.swift]: Probably not. Will remove the test file itself, but leave the project.

**Review by nhamming — DISMISSED** [review]
LGTM

**Review by nhamming — DISMISSED** [review]
+1

File: AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+extension AuthenticationTableViewCell: NibLoadable {}
+
+extension TextButtonTableViewCell: IdentifiableClass {}
+
+fileprivate extension UIAlertController {
+    
+    convenience init(serviceDeletionHandler handler: @escaping () -> Void) {
+        self.init(
+            title: nil,
+            message: NSLocalizedString("Are you sure you want to delete this service?", comment: "Confirmation message for deleting a service"),
```
Comment (ps2) [inline AmplitudeServiceKitUI/AmplitudeServiceTableViewController.swift]: NSLocalizedString -> LocalizedString, and a couple more below.

File: Common/LocalizedString.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+private class FrameworkBundle {
+
+    static let main = Bundle(for: FrameworkBundle.self)
+
+}
+
+func LocalizedString(_ key: String, tableName: String? = nil, value: String? = nil, comment: String) -> String {
+    if let value = value {
+        return NSLocalizedString(key, tableName: tableName, bundle: FrameworkBundle.main, value: value, comment: comment)
```
Comment (ps2) [inline Common/LocalizedString.swift]: NSLocalizedString -> LocalizedString, and one below.

**Review by ps2 — CHANGES_REQUESTED** [review]
Looks good. Just some conversions of NSLocalizedString -> LocalizedString

**Review by ps2 — APPROVED** [review]
🎉 LGTM!

**Pull request merged**